lib/logstash/plugin_mixins/jdbc/statement_handler.rb

```diff
@@ -31,16 +31,20 @@ class NormalStatementHandler < StatementHandler
     # @yieldparam row [Hash{Symbol=>Object}]
     def perform_query(db, sql_last_value, jdbc_paging_enabled, jdbc_page_size)
       query = build_query(db, sql_last_value)
-      if jdbc_paging_enabled
-        query.each_page(jdbc_page_size) do |paged_dataset|
-          paged_dataset.each do |row|
+      # Execute query in transaction cause PG driver require autocommit off for set fetch count
+      # See: https://jdbc.postgresql.org/documentation/head/query.html
+      db.transaction(rollback: :always) do
```

Comment:
I wonder if it might be possible to potentially use [paged_each](https://www.rubydoc.info/github/jeremyevans/sequel/Sequel%2FDataset:paged_each) here? 

If not, do we need to use transactions when paging is not enabled?